games-action/minetest/minetest-5.0.0.ebuild

```diff
@@ -0,0 +1,143 @@
+# Copyright 1999-2019 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit cmake-utils gnome2-utils user
```

Comment:
You can use xdg.eclass instead of gnome2-utils for that icon cache thing. You should then be able to use latest EAPI, 7. 

If you decide to update, bear in mind there are many packages belonging to BDEPEND in here.

Reply:
I want to update to EAPI 7, but I'm having some trouble understanding the difference between BDEPEND and DEPEND. From what I understand, BDEPEND are dependencies required to compile the package on the CBUILD system for the CHOST system, while DEPEND are dependencies required to compile the package on the CHOST system for CHOST system.

I'm not very familiar with the nuance of cross-compilation, so I'm not sure why these two situations would require different dependencies; if a dependency is required to build the package wouldn't it always be necessary regardless of whether the build system is CBUILD or CHOST?

Perhaps an example may help me understand this better. Suppose we have a program called **foo** which dynamically links with a library called **bar**, statically links with a library called **baz**, and is written in a language called _QUX_ (**quxc** is the compiler and **quxpp** is the preprocessor). In EAPI 6, I would have the following defines:

    RDEPEND="bar"
    DEPEND="${RDEPEND} baz quxc quxpp"

In EAPI 7, would this be correct:

    RDEPEND="bar"
    DEPEND="${RDEPEND} baz"
    BDEPEND="quxc quxpp"

Reply:
> I want to update to EAPI 7, but I'm having some trouble understanding the difference between BDEPEND and DEPEND. From what I understand, BDEPEND are dependencies required to compile the package on the CBUILD system for the CHOST system, while DEPEND are dependencies required to compile the package on the CHOST system for CHOST system.
> 
That is correct, however, it doesn't tell much to people who aren't so familiar with package managers technical aspects. 

The way I've learned to make a distinction is: 
BDEPEND is for packages that are only executed while building, and they don't get linked, or don't provide modules/etc needed to build. For example, you can use doxygen to generate documentation files, and this doxygen could be called from any 2nd machine that provides the resources for building (CBUILD) without changing the outcome, so it doesn't need to be present in the CHOST (target machine). Thus, libraries can never be in BDEPEND. 

Your example looked correct to me, so I think you get the idea now :)